Libplanet.Action/ActionEvaluator.cs

```diff
@@ -479,6 +476,64 @@ internal IAccount PrepareInitialDelta(HashDigest<SHA256>? stateRootHash)
             return new Account(new AccountState(_stateStore.GetStateRoot(stateRootHash)));
         }
 
+        internal IReadOnlyList<ICommittedActionEvaluation>
+            ToCommittedEvaluation(
+                IPreEvaluationBlock block,
+                IReadOnlyList<IActionEvaluation> evaluations,
+                HashDigest<SHA256>? baseStateRootHash)
+        {
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            ITrie trie = _stateStore.GetStateRoot(baseStateRootHash);
+            var committedEvaluations = new List<CommittedActionEvaluation>();
+
+            int setCount = 0;
+            foreach (var evaluation in evaluations)
+            {
+                ITrie nextTrie = trie;
+                foreach (var kv in evaluation.OutputState.Delta.ToRawDelta())
+                {
+                    nextTrie = nextTrie.Set(kv.Key, kv.Value);
+                    setCount++;
+                }
+
+                nextTrie = _stateStore.Commit(nextTrie);
+                var committedEvaluation = new CommittedActionEvaluation(
```

Comment:
Originally, `ActionEvaluation.Execute()` was a yield-return method. I am worried about the performance issues with these changes. How about you?

Reply:
Eh, I don't think performance isn't much to worry about. Also, double iteration is necessary down the line as `TxExecution` and `IActionRenderer` have different scopes, so I'd say `IReadOnlyList` is more appropriate here. 🙄

Reply:
Then, can we commit parallelly? I think this makes sense... 

- Locking `IStateStore` is when only writing node.
- Expensive task when committing and calculating `StateRootHash` is organizing tree. 
- And it can be computed in parallel because each `ITrie` does not affect each other.

For sure, I do not think double iteration has an effect on performance. It is already working like that.

Reply:
> Locking IStateStore is when only writing node.

This might be a good idea, but I think locking should be handled from `IStateStore` side not `IActionEvaluator` side.

> Then, can we commit parallelly?

- I don't quite understand this point. `ITrie`'s are independent in the sense that they are all immutable, but they are created sequentially **and** share references to nodes internally.
- Committing/computing state root hash in parallel means having write access to `KeyValueStore` in parallel. Again, I think this should be handled from `IStateStore` side not `IActionEvaluator` side.
- I don't see how committing in parallel would save time. This might be a viable option with `IWorld` later down the line, but even in that case, due to how "separate" `ITrie`'s might be tangled with each other, committing in parallel might not be as easy as you might think.

Reply:
Ok, I might be something overthinking about this. I mean parallelism committing when migrating to `IWorld`. Don't mind what I'm saying. 😓